In [2]:
import pandas as pd
import numpy as np

In [3]:
matches_df = pd.read_csv('../data/results.csv')
rankings_df = pd.read_csv('../data/fifa_ranking.csv')

print("Matches: ", matches_df.shape)
print("Rankings: ", rankings_df.shape)

print("Matches Columns: ", matches_df.columns.tolist())
print("Rankings Columns: ", rankings_df.columns.tolist())

Matches:  (49287, 9)
Rankings:  (67472, 8)
Matches Columns:  ['date', 'home_team', 'away_team', 'home_score', 'away_score', 'tournament', 'city', 'country', 'neutral']
Rankings Columns:  ['rank', 'country_full', 'country_abrv', 'total_points', 'previous_points', 'rank_change', 'confederation', 'rank_date']


#### Checking missing values

In [4]:
print("Missing values in Matches: \n", matches_df.isnull().sum())
print("Missing values in Rankings: \n", rankings_df.isnull().sum())

print(" ")
print("Date range of matches: ")
print("Earliest: ", matches_df['date'].min())
print("Latest: ", matches_df['date'].max())

Missing values in Matches: 
 date           0
home_team      0
away_team      0
home_score    72
away_score    72
tournament     0
city           0
country        0
neutral        0
dtype: int64
Missing values in Rankings: 
 rank               9
country_full       0
country_abrv       0
total_points       0
previous_points    0
rank_change        0
confederation      0
rank_date          0
dtype: int64
 
Date range of matches: 
Earliest:  1872-11-30
Latest:  2026-06-27


##### Since the data ranges from 1872-2026, we only want the recent history to as matches from 1872 are not very useful, football has evolved since then.

In [5]:
#proper date format
matches_df['date'] = pd.to_datetime(matches_df['date'])
print(matches_df['date'].head(3))


0   1872-11-30
1   1873-03-08
2   1874-03-07
Name: date, dtype: datetime64[us]


In [9]:
# keeping only matches from 2014 onwards
matches_df = matches_df[matches_df['date'] >= '2014-01-01']

print("Matches after 2014:")
print(matches_df.shape)
print(matches_df.head())

Matches after 2014:
(11756, 9)
            date home_team away_team  home_score  away_score  \
37531 2014-01-01    Kuwait    Jordan         1.0         2.0   
37532 2014-01-04   Bahrain    Jordan         0.0         1.0   
37533 2014-01-04   Namibia     Ghana         0.0         1.0   
37534 2014-01-04   Nigeria  Ethiopia         2.0         1.0   
37535 2014-01-04     Qatar    Kuwait         3.0         0.0   

              tournament      city  country  neutral  
37531  WAFF Championship      Doha    Qatar     True  
37532  WAFF Championship      Doha    Qatar     True  
37533           Friendly  Windhoek  Namibia    False  
37534           Friendly     Abuja  Nigeria    False  
37535  WAFF Championship      Doha    Qatar    False  


In [12]:
# Seeing tournament types
print(matches_df['tournament'].value_counts())

# adding a feature for friendly matches
matches_df['is_friendly'] = matches_df['tournament'] == 'Friendly'

print(" ")
print(matches_df['is_friendly'].value_counts())

tournament
Friendly                                3402
FIFA World Cup qualification            2618
African Cup of Nations qualification     858
UEFA Euro qualification                  768
UEFA Nations League                      658
                                        ... 
Mukuru 4 Nations                           2
Benedikt Fontana Cup                       1
ConIFA Challenger Cup                      1
CONMEBOL–UEFA Cup of Champions             1
South Asian Super Cup                      1
Name: count, Length: 85, dtype: int64
 
is_friendly
False    8354
True     3402
Name: count, dtype: int64


### Getting the World cup 2026 teams

In [13]:
world_cup_2026_teams = [

    # Hosts (CONCACAF)
    'Canada',
    'Mexico',
    'United States',

    # AFC (Asia)
    'Australia',
    'Iraq',
    'Iran',
    'Japan',
    'Jordan',
    'South Korea',
    'Qatar',
    'Saudi Arabia',
    'Uzbekistan',

    # CAF (Africa)
    'Algeria',
    'Cabo Verde',
    'Democratic Republic of Congo',
    "Côte d'Ivoire",
    'Egypt',
    'Ghana',
    'Morocco',
    'Senegal',
    'South Africa',
    'Tunisia',

    # CONCACAF
    'Curaçao',
    'Haiti',
    'Panama',

    # CONMEBOL (South America)
    'Argentina',
    'Brazil',
    'Colombia',
    'Ecuador',
    'Paraguay',
    'Uruguay',

    # OFC
    'New Zealand',

    # UEFA (Europe)
    'Austria',
    'Belgium',
    'Bosnia and Herzegovina',
    'Croatia',
    'Czechia',
    'England',
    'France',
    'Germany',
    'Netherlands',
    'Norway',
    'Portugal',
    'Scotland',
    'Spain',
    'Sweden',
    'Switzerland',
    'Türkiye'
]

print(f"Total teams: {len(world_cup_2026_teams)}")

Total teams: 48


#### Filtering matches to only World Cup teams

In [15]:
wc_matches = matches_df[
    matches_df['home_team'].isin(world_cup_2026_teams) &
    matches_df['away_team'].isin(world_cup_2026_teams)
]

print(wc_matches.shape)
print(wc_matches.head())

(1298, 10)
            date      home_team    away_team  home_score  away_score  \
37537 2014-01-07          Qatar       Jordan         2.0         0.0   
37547 2014-01-29         Mexico  South Korea         4.0         0.0   
37549 2014-02-01  United States  South Korea         2.0         0.0   
37560 2014-03-05      Australia      Ecuador         3.0         4.0   
37561 2014-03-05        Austria      Uruguay         1.0         1.0   

              tournament         city        country  neutral  is_friendly  
37537  WAFF Championship         Doha          Qatar    False        False  
37547           Friendly  San Antonio  United States     True         True  
37549           Friendly       Carson  United States    False         True  
37560           Friendly       London        England     True         True  
37561           Friendly   Klagenfurt        Austria    False         True  


In [16]:
wc_matches.to_csv('../data/wc_matches.csv', index=False)